library/std/src/sys/windows/thread_local_key.rs

```diff
@@ -1,14 +1,49 @@
 use crate::cell::UnsafeCell;
 use crate::ptr;
 use crate::sync::atomic::{
-    AtomicPtr, AtomicU32,
+    AtomicBool, AtomicPtr, AtomicU32,
     Ordering::{AcqRel, Acquire, Relaxed, Release},
 };
 use crate::sys::c;
 
 #[cfg(test)]
 mod tests;
 
+/// An optimization hint. The compiler is often smart enough to know if an atomic
```

Comment:
I'm surprised the compiler can do this! That's an improvement -- it definitely didn't used to be able to.

Reply:
Aye. In my tests it seems to work well for simple local-ish cases. Dunno if it'd hold up in more complex cases tho.